plugins/Diagnostics/Diagnostic/Diagnostic.php

```diff
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\Diagnostics\Diagnostic;
+
+/**
+ * Performs a diagnostic on the system or Piwik.
```

Comment:
> Nothing specific. Regarding translations: the diagnostic results take a label, so it's just a string and the user can use the translator or not. I think we should document when a translation key is expected (because it's a particular case) but not the other way around (because it's the default/normal case).

What I meant in this case is, is it recommended to use a translation?

> I'm not sure I understand, these tools cool be implemented as diagnostics too

Can you elaborate on how? Would they be output in the System Check page/run in diagnostics:run?